test/data-basic.jl

```diff
@@ -35,6 +35,35 @@
         @test isapprox(result["objective"], 16551.7; atol=1e0)
     end
 
+    @testset "In-place make_basic_network!" begin
+        data = PowerModels.parse_file("../test/data/matpower/case7_tplgy.m")
+        data_ = deepcopy(data)
+        
+        # Test #1: `make_basic_network` is non-destructive
+        data_basic = PowerModels.make_basic_network(data)
+        @test data == data_        # input data not modified
+        @test data_basic !== data  # we created a new dict
```

Comment:
These ones check that we did not modify the input data, and that we returned a new `Dict` object